Sources/EventProducer/TidalEventSender.swift

```diff
@@ -118,6 +118,11 @@ public final class TidalEventSender: EventSender {
 		)
 		endOutage(eventName: event.name)
 	}
+	
+	public func sendAllEvents() async throws {
+		let headerHelper = HeaderHelper(credentialsProvider: config?.credentialsProvider)
+		try await scheduler?.sendAllEvents(headerHelper: headerHelper)
+	}
```

Comment:
Let's see what others think too, but should we try to recover form an expired token error, and retry?
I think auth errors should be handled internally.

```
 catch let EventProducerError.unauthorized(code) {
			_ = try await headerHelper.credentialsProvider?.getCredentials(apiErrorSubStatus: code.description)
			try await scheduler?.sendAllEvents(headerHelper: headerHelper)
} catch {
    throw error
}
```

Reply:
cc @ceciliasaraiva @e-kononenko

Reply:
Although this is not too time critical, this is a great addition, love it! 🙌 great to handle this edge case 🚀

Reply:
I don't have context over all the auth error handling, it would be great at some point to have some consistent behavior in whole sdk. 🤓
As for this specific logic, would it depend on the code received to decide if we retry again or not? Otherwise, simply retrying it every time would also work. And maybe we actually log that error in the first catch from the suggestion - which we won't be aware of in case the retry succeeds.

Reply:
As user would already be aware of this error on client side (they have no auth), maybe we don't have to notify them ourselves, a client would do that in more evident manner.
Or did you mean to just log it to console for a more transparent approach?

Bare in mind, this will hopefully happen when the app is not being used at all anymore.

Reply:
Hmm, I don't mean necessarily notifying. My question which was not clear, sorry for that, is that maybe our decision to directly retry or not could be decided upon code received from `EventProducerError.unauthorized(code)` - since I don't have whole context, I'm not sure if this makes sense or not, you will know more.
And then my other idea was to log it here for example:
```
 catch let EventProducerError.unauthorized(code) {
    // maybe we log here
    _ = try await headerHelper.credentialsProvider?.getCredentials(apiErrorSubStatus: code.description)
    try await scheduler?.sendAllEvents(headerHelper: headerHelper)
} catch {
    throw error
}
```

Reply:
Oh I see. Might be interesting.
I can try and find out if there are defined codes for this that we can filter out, but I believe we just use that to forward the value and do not have a `specified` list we can lean on. Checking!

Logging wise, sure, sounds good!! 🙌